src/chainstate/burn/db/burndb.rs

```diff
@@ -843,674 +648,811 @@ impl BurnDB {
-    pub fn get_block_burn_amount<'a>(ic: &BurnDBConn<'a>, block_height: u64, tip_block_hash: &BurnchainHeaderHash) -> Result<u64, db_error> {
-        assert!(block_height < BLOCK_HEIGHT_MAX);
+        let stable_snapshot = db_handle.get_block_snapshot_by_height(stable_block_height)?
+            .ok_or_else(|| {
+                // shouldn't be possible, but don't panic since this is network-callable code
+                error!("Failed to load snapshot for block {} from fork {}", stable_block_height, &chain_tip.burn_header_hash);
+                db_error::Corruption
+            })?;
 
-        let user_burns = BurnDB::get_user_burns_by_block(ic, block_height, tip_block_hash)?;
-        let block_commits = BurnDB::get_block_commits_by_block(ic, block_height, tip_block_hash)?;
-        let mut burn_total : u64 = 0;
-        
-        for i in 0..user_burns.len() {
+        // get all consensus hashes between the chain tip, and the stable height back
+        // MAX_NEIGHBOR_BLOCK_DELAY
+        let oldest_height = 
+            if stable_snapshot.block_height < MAX_NEIGHBOR_BLOCK_DELAY {
+                0
+            }
+            else {
+                stable_snapshot.block_height - MAX_NEIGHBOR_BLOCK_DELAY
+            };
+
+        let mut last_consensus_hashes = HashMap::new();
+        for height in oldest_height..chain_tip.block_height {
+            let ch = db_handle.get_consensus_at(height)?
+                .unwrap_or(ConsensusHash::empty());
+            last_consensus_hashes.insert(height, ch);
+        }
+
+        test_debug!("Chain view: {},{}-{},{}", chain_tip.block_height, chain_tip.consensus_hash, stable_block_height, stable_snapshot.consensus_hash);
+        Ok(BurnchainView {
+            burn_block_height: chain_tip.block_height, 
+            burn_consensus_hash: chain_tip.consensus_hash,
+            burn_stable_block_height: stable_block_height,
+            burn_stable_consensus_hash: stable_snapshot.consensus_hash,
+            last_consensus_hashes: last_consensus_hashes
+        })
+    }
+}
+
+pub struct PoxDB;
+
+impl PoxDB {
+    /// Get the canonical PoX identifier for a given burnchain header hash.
+    ///   this result may change if a previously unknown PoX anchor is processed.
+    pub fn get_canonical_pox_id(&self, _burnchain_header_hash: &BurnchainHeaderHash) -> Result<PoxIdentifier, ()> {
+        Ok(PoxIdentifier::stubbed())
+    }
+    /// Get the parent PoX identifier for a given identifier.
+    ///  If the PoX identifier does not have a parent (should only be true for the "base" PoX identifier)
+    ///    return an error.
+    pub fn get_parent_pox_id(&self, _pox_id: &PoxIdentifier) -> Result<PoxIdentifier, ()> {
+        Err(())
+    }
+    /// does the given pox identifier describe a child fork of the parent pox identifier? this returns true if they
+    ///   are equal or child is a descendant.
+    pub fn is_pox_id_a_child(&self, parent: &PoxIdentifier, child: &PoxIdentifier) -> bool {
```

Comment:
Minor nit -- recommend not using articles ("a", "an", "the") in method names.  Maybe this method could be called "is_pox_id_descendant"?

Reply:
Sure -- I'll rename to `descends_from` as that seems a clearer description of the condition being evaluated.